### DISCUSSION TITLE
Relationship triggers do not cascade delete when parent collection is deleted (m2a)

### DISCUSSION THREAD

**Discussion by frosty5689** [directus/directus#11245] [Archive]
Hi

I have a m2a relationship setup and I want to cascade delete the children when the parent gets deleted

![image](https://user-images.githubusercontent.com/4147147/150870610-37379f19-bf59-4e13-a382-5d1bbdd3763e.png)

I set the relationship triggers per above screenshot. But when I delete the parent, the child stays behind. Am I misunderstanding something about what this is supposed to do? Or is this a bug?

Thanks

**Comment by eeprom** [comment]
I'm also experiencing this when using M2A relationships. Could any clarify if this is expected, because I'd expect the cascading delete to remove children.

**Comment by rijkvanzanten** [comment]
> But when I delete the parent, the child stays behind.

I think the confusion here might be around what the "parent" and "child" are. Are you looking to delete the related row? The cascade option here will cascade delete on the junction row, not the related item 👍🏻

**Comment by w0kyj** [comment]
Client reported:
Deleting a related item:
- No notification that it is related
- Cascade clean up also not happening

Would like to have cascade notifications and clean up options for this complex relationship structure. Support would be similar to more traditional M2M cascade features.

**Comment by gabrielmustiere** [comment]
Hello,

I join the discussion and I agree with what was said above, I also expected to have at first the links removed but also the linked objects. Is it planned? Would you have a solution to implement it easily, using flows maybe?

Need open an issue ?

Txs

**Comment by jagyas** [comment]
As a workaround, when I am creating a many to any (M2A) between parent and children, I am also creating a many to one (M2O) between children and parent (not in junction) so that I can trigger cascade delete from parent to children. The only problem is once I have created parent, I have to edit it again and link it with children manually.
